refnx/reflect/simulation.py

```diff
@@ -0,0 +1,167 @@
+        self.layers = np.zeros((self.av_layers.shape[0],
+                                self.av_layers.shape[1],
+                                len(self.u.trajectory)))
+        self.layers[:, 0, :] = layer_thickness
+        self.layers[:, 3, :] = layer_thickness * roughness
+        self.layers[:, 4, :] = 0
+        self.av_layers[:, 0] = layer_thickness
+        self.av_layers[:, 3] = layer_thickness * roughness
+        self.av_layers[:, 4] = 0
+        self.flip = flip
+        self.cut_off = cut_off
+        self.get_sld_profile()
+        if verbose:
+            print('SLD profile determined.')
+
+    def read_pdb(self):
+        """Parse pdb file.
+
+        This reads the pdb file into memory.
+        """
+        self.u = mda.Universe(self.pdbfile)
+
+    def read_lgt(self):
+        """Parses .lgt.
+
+        Parses the lgtfile. If no lgtfile is defined falass will help the user
+        to build one by working through the
+        atom types in the pdb file and requesting input of the real and
+        imaginary scattering lengths. This will also
+        occur if a atom type if found in the pdbfile but not in the given lgts
+        file. falass will write the lgtfile
+        to disk if atom types do not feature in the given lgtfile or one is
+        written from scratch.
+        """
+        if self.lgtfile:
+            file = open(self.lgtfile, 'r')
+            for i, line in enumerate(file):
+                line_list = line.split()
+                i = 1
+                if self.xray:
+                    i *= 2.817940
+                self.scatlens[line_list[0]] = [float(line_list[1]) * i,
+                                               float(line_list[2]) * i]
+            file.close()
+        else:
+            raise ValueError("No lgtfile has been defined.")
+
+    def get_sld_profile(self):
+        """Calculate SLD profile.
+
+        This will calculate the SLD profile for each of the timesteps. This is
+        achieved by summing the scattering lengths for each of the atoms found
+        in a given layer (of defined thickness). The total scattering length is
+        converted to a density by division by the volume of the layer.
+        """
+        u = self.u
+        k = 0
+        for ts in u.trajectory:
+            for atom in range(0, len(u.atoms)):
+                analysis_layers = self.layers.shape[0] * self.layers[0, 0, k]
+                if u.atoms[atom].position[2] < analysis_layers:
+                    bin_choose = int(u.atoms[atom].position[2] /
+                                     self.layers[0, 0, k])
+                    self.layers[bin_choose, 1, k] += \
+                        self.scatlens[u.atoms[atom].name][0]
+                    self.layers[bin_choose, 2, k] += \
+                        self.scatlens[u.atoms[atom].name][1]
+            self.layers[:, 1, k] /= (u.dimensions[0] * u.dimensions[1] *
+                                     self.layers[0, 0, k])
```

Comment:
Is the volume fraction of solvent taken into account somehow, or is there explicit solvent in the box?

Reply:
there is explicit solvent in the simulation (and there probably should be in most simulations). The solvent is just acmw (hence the zero scattering in the .lgts file.